tools/manifest/sourcefile.py

```diff
@@ -212,54 +247,66 @@ def __getstate__(self):
         return rv
 
     def name_prefix(self, prefix):
+        # type: (bytes) -> bool
         """Check if the filename starts with a given prefix
 
         :param prefix: The prefix to check"""
         return self.name.startswith(prefix)
 
     def is_dir(self):
+        # type: () -> bool
         """Return whether this file represents a directory."""
         if self.contents is not None:
             return False
 
         return os.path.isdir(self.rel_path)
 
     def open(self):
+        # type: () -> BinaryIO
         """
         Return either
         * the contents specified in the constructor, if any;
         * a File object opened for reading the file contents.
         """
 
         if self.contents is not None:
-            file_obj = ContextManagerBytesIO(self.contents)
+            wrapped = ContextManagerBytesIO(self.contents)
+            if MYPY:
+                file_obj = cast(BinaryIO, ContextManagerBytesIO)
+            else:
+                file_obj = wrapped
         else:
             file_obj = open(self.path, 'rb')
         return file_obj
 
     @cached_property
     def path(self):
+        # type: () -> Union[bytes, Text]
         return os.path.join(self.tests_root, self.rel_path)
 
     @cached_property
     def rel_url(self):
+        # type: () -> Text
         assert not os.path.isabs(self.rel_path), self.rel_path
         return self.rel_path.replace(os.sep, "/")
 
     @cached_property
     def url(self):
+        # type: () -> Text
         return urljoin(self.url_base, self.rel_url)
 
     @cached_property
     def hash(self):
+        # type: () -> bytes
         if not self._hash:
             with self.open() as f:
                 content = f.read()
-            data = "".join(("blob ", str(len(content)), "\0", content))
+            data = b"".join((b"blob ", bytes(len(content)), b"\0", content))
```

Comment:
```
$ python3
>>> bytes(7)
b'\x00\x00\x00\x00\x00\x00\x00'
```

Reply:
WE DON'T HAVE ANY TESTS THAT FAIL, THIS WORKS FINE. /s